### DISCUSSION TITLE
Using an alternative integrator

### DISCUSSION THREAD

**Discussion by msmommer** [OpenModelica/OpenModelica#12762] [General]
Hi,

is there a protocol or interface one could use to add to openmodelica a custom integrator? Concretely, if I had some exotic integrator, adapt it so that I can use it instead of DASSL.

I think I remember seeing something like that but I have failed to find the document explaining how to do that.

Any pointer would be greatly appreciated!

Regards, and thanks,

          Mario

**Comment by casella** [comment]
@msmommer the easy way to do that is to export your model as an FMU-ME, and then link your exotic integrator to the state-space description it provides.

Of course you can also add a new solver to the runtime, but that requires to get a bit into it. If you want to try, you can ask @AnHeuermann for some help.

**Reply by AnHeuermann** [reply]
Building a FMU importing tool that can simulate a Model Exchange FMU with your desired solver is probably the best way to go. You could build on top of some existing importing tool, maybe from the Julia or Python community.

If your exotic solver is an explicit or implicit Runge-Kutta method it would be (relativ) easy to add the butcher table to our C runtime and use GBODE with that RK method.
Otherwise the process of adding new solvers to OpenModelica is very involved, see for example https://github.com/OpenModelica/OpenModelica/pull/12278 where some user want's to add IDA to Co-Simulation FMUs.
